src/Storefront/Resources/app/storefront/src/plugin/cookie/cookie-configuration.plugin.js

```diff
@@ -145,12 +150,206 @@ export default class CookieConfiguration extends Plugin {
+                headers: {
+                    'X-Requested-With': 'XMLHttpRequest',
+                },
+            });
+
+            return await response.json();
+        } catch (error) {
+            console.error('Failed to fetch cookie groups:', error);
+            return null;
+        }
+    }
+
+    /**
+     * Check if cookie configuration hash has changed and reset cookies if needed
+     * @private
+     */
+    async _checkCookieConfigurationHash() {
+        const { cookiePreference, cookieConfigHash } = this.options;
+        const hasPreference = CookieStorage.getItem(cookiePreference);
+        const storedHash = CookieStorage.getItem(cookieConfigHash);
+
+        if (!hasPreference && !storedHash) {
+            return;
+        }
+
+        const data = await this._fetchCookieGroups();
+        if (!data) {
+            return; // Error already logged in _fetchCookieGroups
+        }
+
+        const currentHash = data.hash;
+
+        if (storedHash && storedHash !== currentHash) {
+            await this._resetCookieConfiguration(data);
+        }
+
+        CookieStorage.setItem(cookieConfigHash, currentHash, '30');
+    }
+
+
+    /**
+     * Reset cookie configuration when hash has changed
+     * @private
+     */
+    async _resetCookieConfiguration(data) {
+        const cookieGroups = data.elements || [];
+        const { activeCookieNames, inactiveCookieNames } = this._setTechnicallyRequiredCookiesAndGetState(cookieGroups);
+
+        CookieStorage.removeItem(this.options.cookiePreference);
+
+        const updatedActiveCookieNames = activeCookieNames.filter(name => name !== this.options.cookiePreference);
+        this._handleUpdateListener(updatedActiveCookieNames, inactiveCookieNames);
+
+        this._checkAndShowCookieBarIfNeeded();
+    }
+
+    /**
+     * Extract all cookie names from cookie groups
+     * @private
+     */
+    _getAllCookieNamesFromGroups(cookieGroups) {
+        return cookieGroups
+            .flatMap(group => group.entries ? group.entries.map(entry => entry.cookie) : (group.cookie ? [group.cookie] : []))
+            .filter(cookieName => cookieName);
+    }
+
+    /**
+     * Get technically required cookie names that are managed by PHP
+     * These cookies should not be set by JavaScript
+     * @private
+     */
+    _getTechnicallyRequiredCookieNames() {
+        return ['session-', 'timezone'];
+    }
+
+    /**
+     * Get technically required cookies that should be set (excludes PHP-managed ones)
+     * @param {Array} cookieGroups - Array of cookie groups from API
+     * @returns {Array} Array of cookie objects that should be set
+     * @private
+     */
+    _getTechnicallyRequiredCookiesToSet(cookieGroups) {
+        const phpManagedCookies = this._getTechnicallyRequiredCookieNames();
+        const cookiesToSet = [];
+
+        for (let i = 0; i < cookieGroups.length; i++) {
+            const group = cookieGroups[i];
+            if (group.isRequired) {
```

Comment:
Could `group.isRequired` be an early return? And potentially more? At the moment its 4 level ifs.